### DISCUSSION TITLE
Best practise for non-page files in routes/

### DISCUSSION THREAD

**Discussion by jEnbuska** [remix-run/remix#4289] [Q&A] [answered]
What is the best practice for defining files inside `routes/`, that should not be rendered as a page?

I'm asking since after spending a few hours getting to know remix, I didn't find examples or documentation for solving this problem.

In my experience, both: Not having a clear convention for file structure and having a convention where related functionalities are far from where they are needed, lead in practise to just huge page files, and little code reuse.

In single-page apps, I have usually prefixed the non page files and folders with some special characters, so it's easier to differ the page files from the other utilities

**Comment by kiliman** [comment, marked as answer]
Ideally, you would include non-route files outside of the *routes* folder. You can them import them like `import Button from '~/components/Button'`.

If you want to colocate them near your route file, then you'll need to update *remix.config* and set `ignoredRouteFiles` with patterns to ignore[1]. Something like: `['**/*.component.tsx', '**/*.css', ...]`

Another option is to look at `remix-flat-routes`[2]. It supports colocating files directly in the route folder. See section on *Colocation*[3]

[1] https://remix.run/docs/en/v1/api/conventions#ignoredroutefiles
[2] https://github.com/kiliman/remix-flat-routes
[3] https://github.com/kiliman/remix-flat-routes#colocation

**Reply by jEnbuska** [reply]
Awesome! Thanks for the quick response. I think for my use case I will be using the `ignoreRouteFiles` pattern.

**Reply by richardscarrott** [reply]
Just for anybody else who lands here, remix v2 routing supports colocation out of the box https://remix.run/docs/en/main/file-conventions/route-files-v2#folders-for-organization
